### PR TITLE
Restructure test fixtures/utilities

### PR DESCRIPTION
**Pull request by dominikschubert**
Some cleanup, restructuring of our pytest setup.

Mostly extracting/moving things around.

This PR intentionally tries to avoid touching any logic and doesn't move actual test files. (some tests had to be fixed/removed/skipped though)

### CODE REVIEW

File: localstack/utils/testing/pytest/snapshot_plugin.py [hunk 1]
```diff
@@ -39,7 +49,7 @@ def pytest_runtest_call(item: Item) -> None:
 
 @pytest.fixture(name="account_id", scope="session")
 def fixture_account_id():
-    sts_client = _client("sts")
+    sts_client = _client("sts")  # TODO: fix
```
Comment (thrau) [inline localstack/utils/testing/pytest/snapshot_plugin.py:52]: fix what? :)

File: tests/integration/test_sts.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def test_custom_caller_identity(self, sts_client, iam_client, monkeypatch):
         assert response["User"]["UserName"] == test_name
         assert response["User"]["UserId"] == test_id
 
-    @only_localstack
+    @pytest.mark.only_localstack
```
Comment (thrau) [inline tests/integration/test_sts.py:185]: much better!

File: tests/integration/test_cloudformation.py [hunk 3]
```diff
@@ -2334,134 +1647,125 @@ def test_cfn_with_apigateway_resources(self):
...
-        # Delete the stack and wait for the status 'DELETE_COMPLETE' of the stack
-        delete_and_await_stack(stack_name)
-
-    def test_cfn_with_multiple_route_tables(self):
-        ec2_client = aws_stack.create_external_boto_client("ec2")
-
+    # TODO: this test seems a bit convoluted without a clear goal
+    # def test_updating_stack_with_iam_role(self):
+    #     lambda_client = aws_stack.create_external_boto_client("lambda")
+    #     iam = aws_stack.create_external_boto_client("iam")
```
Comment (thrau) [inline tests/integration/test_cloudformation.py:1678]: i think we should either remove the test all together if it adds no value or is covered by other tests, or simply skip it if we want to keep the code for some reason. we shouldn't keep code in the repo that commented out where it's not clear what to do with it.

**Review by thrau — COMMENTED** [review]
looks great overall, thanks for cleaning up :100:  just a couple of comments and nitpicks about naming.

* generally the goal is to move more complex systems out of `localstack.utils`, so i would love to see the testing package moved from `localstack/utils/` to `localstack/` as we discussed.
* i think it would be better to rename `fixture_plugin.py` and `snapshot_plugin.py` to `fixtures.py`, and `snapshot.py`. i don't think the `_plugin` adds any useful information. everything in pytest is kinda a plugin.

File: localstack/utils/testing/pytest/snapshot_plugin.py (hunk 1 above)
Comment (dominikschubert) [reply localstack/utils/testing/pytest/snapshot_plugin.py]: extracting _client :+1:  added a longer comment

File: tests/integration/test_cloudformation.py (hunk 3 above)
Comment (dominikschubert) [reply tests/integration/test_cloudformation.py]: Added a slightly refactored version back but kept a todo to re-evaluate. There a few tests in there that we might be able to drop soon :tm:

**Review by thrau — APPROVED** [review]
:rocket:

**Pull request merged**